components/director/internal/domain/tenant/service.go

```diff
@@ -341,7 +374,17 @@ func (s *labeledService) UpsertSingle(ctx context.Context, tenantInput model.Bus
 	return s.upsertTenant(ctx, tenantInput, s.tenantMappingRepo.Upsert)
 }
 
-func (s *labeledService) upsertTenant(ctx context.Context, tenantInput model.BusinessTenantMappingInput, upsertFunc func(context.Context, model.BusinessTenantMapping) error) (string, error) {
+func (s *labeledService) upsertTenant(ctx context.Context, tenantInput model.BusinessTenantMappingInput, upsertFunc func(context.Context, model.BusinessTenantMapping) (string, error)) (string, error) {
+	parents, err := s.ListsByExternalIDs(ctx, tenantInput.Parents)
+	if err != nil {
+		return "", errors.Wrap(err, "while listing tenants by external ids")
+	}
+	parentInternalIDs := make([]string, 0, len(parents))
+	for _, parent := range parents {
+		parentInternalIDs = append(parentInternalIDs, parent.ID)
```

Comment:
This is not covered. Maybe just an adaptation in the already existing unit test for the `UpsertSingle` function will be enough.